framework/src/utils/ADUtils.C

```diff
@@ -0,0 +1,53 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADUtils.h"
+
+#include "SystemBase.h"
+#include "libmesh/system.h"
+#include "libmesh/dof_map.h"
+
+namespace Moose
+{
+
+std::unordered_map<dof_id_type, Real>
+globalDofIndexToDerivative(const ADReal & ad_real,
+                           const SystemBase & sys,
+                           const Elem * elem,
+                           const ElementType elem_type /*=ElementType::Element*/)
+{
```

Comment:
What do we anticipate this being used for in the future? Did you have anything in mind?

Reply:
See #15450